### ISSUE TITLE
请教lora微调的时候loss一直是0

### ISSUE THREAD

**Issue by xiang-hui744** [zjunlp/KnowLM#139]
<img width="1235" alt="image" src="https://github.com/zjunlp/KnowLM/assets/57001862/7ee3b655-573a-4321-a3c4-614986de2b92">

**Comment by xiang-hui744** [comment]
到后面loss又猛增
{'loss': 0.0, 'learning_rate': 2.9999999999999997e-05, 'epoch': 0.03}                                                                                                
{'loss': 0.0, 'learning_rate': 5.9999999999999995e-05, 'epoch': 0.07}                                                                                                
{'loss': 0.0, 'learning_rate': 8.999999999999999e-05, 'epoch': 0.1}                                                                                                  
{'loss': 0.0, 'learning_rate': 0.00011999999999999999, 'epoch': 0.14}                                                                                                
{'loss': 0.0, 'learning_rate': 0.00015, 'epoch': 0.17}                                                                                                               
{'loss': 0.0, 'learning_rate': 0.00017999999999999998, 'epoch': 0.2}                                                                                                 
{'loss': 0.0, 'learning_rate': 0.00020999999999999998, 'epoch': 0.24}                                                                                                
{'loss': 0.0, 'learning_rate': 0.00023999999999999998, 'epoch': 0.27}                                                                                                
{'loss': 0.0, 'learning_rate': 0.00027, 'epoch': 0.31}                                                                                                               
{'loss': 0.0, 'learning_rate': 0.0003, 'epoch': 0.34}                                                                                                                
{'eval_loss': nan, 'eval_runtime': 73.2303, 'eval_samples_per_second': 6.828, 'eval_steps_per_second': 0.86, 'epoch': 0.34}                                          
{'loss': 0.0, 'learning_rate': 0.0002977777777777777, 'epoch': 0.38}                                                                                                                                                         
{'loss': 1992428617728.0, 'learning_rate': 0.0002955555555555555, 'epoch': 0.41}                                                                                                                                             
  8%|██████████▎                                                                                                                 | 12/145 [33:20<6:37:27, 179.30s/it]

**Comment by MikeDean2367** [comment]
您好，请提供一下运行的命令和数据格式

**Comment by xiang-hui744** [comment]
> 您好，请提供一下运行的命令和数据格式
命令：CUDA_VISIBLE_DEVICES=0 accelerate launch finetune.py --data_path ./data/train.json --base_model /home/keke/KnowLM/knowlm-13b-zhixi

数据有8000条，格式：
<img width="1395" alt="image" src="https://github.com/zjunlp/KnowLM/assets/57001862/08af0526-bd78-43e8-9559-e5d9b04f5c91">

**Comment by xiang-hui744** [comment]
> > 您好，请提供一下运行的命令和数据格式
> > 命令：CUDA_VISIBLE_DEVICES=0 accelerate launch finetune.py --data_path ./data/train.json --base_model /home/keke/KnowLM/knowlm-13b-zhixi
> 
> 数据有8000条，格式： <img alt="image" width="1395" src="https://private-user-images.githubusercontent.com/57001862/342069516-08af0526-bd78-43e8-9559-e5d9b04f5c91.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTkxNTExOTIsIm5iZiI6MTcxOTE1MDg5MiwicGF0aCI6Ii81NzAwMTg2Mi8zNDIwNjk1MTYtMDhhZjA1MjYtYmQ3OC00M2U4LTk1NTktZTVkOWIwNGY1YzkxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjIzVDEzNTQ1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU3NTY4MzZkZGUzY2NhMjE1OTM0YTcwNzEwODdhYzA0OGI4YmE2ZDMxMWRiNDM3MzY2NDZiMTVjNzk5ZTA1YjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Jq1O8wDJVPzW29l6x4Ks8E3Q2d_uzEuSvMGhnxEq1GU">
其中一条样本如下：
{
        "instruction": "我希望你根据关系列表从给定的输入中抽取所有可能的关系三元组，并以(头实体,关系,尾实体)的形式回答，relation可从列表[因果，时序]中选取。head和tail从列表['立案侦查', '逮捕']中选取",
        "input": "成都市原市长助理门生涉嫌受贿案侦查终结，移送审查起诉_一号专案_澎湃新闻-ThePape。\n\n中新网成都1月5日电，5日，成都市人民检察院官方网站公布了近段时间成都市检察机关查处的4起职务犯罪案件信息。其中，原成都市市长助理、秘书长门生（副厅级）涉嫌受贿一案，由成都市人民检察院侦查终结，移送审查起诉。门生，男，汉族，1959年3月生，四川内江人。2009年门生因抗震救灾有功等原因，兼任成都市市长助理。2013年2月，门生转任成都市市长助理、市政府秘书长，并兼市政府办公厅党组书记，市委党校（行政学院）第一副校（院）长等职迄今。2015年06月10日，成都市纪委发布消息称，经成都市委同意，成都市纪委对成都市市长助理、市政府秘书长门生涉嫌严重违纪违法问题进行立案调查。门生是2014年8月以来，成都继陈争鸣、刘俊林及周鸿德之后落马的第4位市长助理。2015年9月17日，成都市人民检察院依法决定对成都市政府原市长助理、秘书长门生（副厅级）受贿立案侦查。2015年9月25日，经四川省人民检察院决定，依法对门生予以逮捕。此外，成都市检察院还公布了成都武侯区投资促进局副局长岳文婷、成都锦江区三圣街道办事处办公室原主任胡杨、成都市公积金管理中心原党委书记胡旭光的职务犯罪案件信息。",
        "output": "['(立案侦查,时序,逮捕)']"
    },

**Comment by MikeDean2367** [comment]
您好，我将您的instruction+input全部拼接起来后，发现分词完成后的token数量为865，根据您提供的截图，设置的最大长度是512。因此建议您增加最大长度来解决这个问题。如仍有问题，请告知我 :)
